packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +46,371 @@ import {Heading, Text} from '@react-spectrum/typography';
+          </ButtonGroup>
+          <Content>
+            <Form>
+              <TextField label="Name" />
+              <Checkbox>Make private</Checkbox>
+            </Form>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Additionally, DialogTrigger accepts an `onOpenChange` prop which is triggered whenever the Dialog is opened or closed. 
+For more information on the `onOpenChange` prop see [DialogTrigger page](TODO LINK HERE).
+
+### Dismissable Dialogs
+
+Dismissable Dialogs support a `onDismiss` prop which is triggered whenever the Dialog's close button is clicked.
+Similar to non-dismissable dialogs, you must chain the DialogTrigger's `close` function with whatever callback you provide as
+`onDismiss`.
 
-*If the component supports validation props, include this section.*
+```tsx example
+function Example() {
+  let alertDismiss = (cb) => {
+    cb();
+    alert('Dialog dismissed.');
+  }
+
+  return (
+    <DialogTrigger isDismissable>
+      <ActionButton>Info</ActionButton>
+      {(close) => (
+        <Dialog onDismiss={() => alertDismiss(close)}>
+          <Heading>Version Info</Heading>
+          <Divider />
+          <Content>
+            <Text>
+              Version 1.0.0, Copyright 2020
+            </Text>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Note that `onDismiss` is optional. If you don't need to add a `onDismiss` handler to your dismissable Dialog, you may omit
+the wrapping `close` function surrounding the Dialog. An example of this can be found in the Examples section [above](#examples).
 
 ## Props
 
 <PropTable component={docs.exports.Dialog} links={docs.links} />
 
 ## Visual Options
 
-*Show examples of all variants and visual props here with links to the design website for more usage details. Examples can be grouped together for conciseness.*
+### Dialog Variants
+
+Dialogs can be rendered as modals, popovers, or trays. See the [DialogTrigger docs](TODO link) for more information.
+
+```tsx example
+<DialogTrigger type="modal">
+  <ActionButton>Trigger Modal</ActionButton>
+  <Dialog>
+    <Heading>Modal</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a modal.
+      </Text>
+    </Content>    
+  </Dialog>
+</DialogTrigger>
+
+<DialogTrigger type="popover">
+  <ActionButton>Trigger Popover</ActionButton>
+  <Dialog>
+    <Heading>Popover</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover.
+      </Text>
+    </Content>    
+  </Dialog>
+</DialogTrigger>
+
+<DialogTrigger type="tray">
+  <ActionButton>Trigger Tray</ActionButton>
+  <Dialog>
+    <Heading>Tray</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a tray.
+      </Text>
+    </Content>    
+  </Dialog>
+</DialogTrigger>
+```
+
+### Size
+
+Only `modal` type Dialogs support a user defined size prop. Note that the `fullscreen` and `fullscreenTakeover` sizes 
```

Comment:
We keep saying size, but do we mean width? Do we need to say that all modals are the same size on mobile/phone devices?

Reply:
Nah, the sizes affect the width and height of the Dialog in some cases (fullscreen/fullscreen takeover), also feels more consistent with the prop name. I can mention the bit about size being the same on mobile/phone devices